File: FrontOffice/src/app/entities/Post.ts
import { Topic } from "./Topic";
import { Client } from "./Client";

export class Post {
    id_post: any;
    content: any;
    likes: any;
    dislikes: any;
    creation_date: Date;
    modified: any;
    user_id: any;
    user: Client;
    topic: Topic;
    comments: Comment[];
  }
  

File: BackEnd/src/main/java/com/example/pifinance_back/Entities/Post.java
package com.example.pifinance_back.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Post implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPost;
    private String content;
    private Integer likes;
    private Integer dislikes;

    @Temporal(TemporalType.TIMESTAMP)
    private Date creationDate;
    private Boolean modified;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "user_id")
    private Client user;
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "topic_id")
    private Topic topic;
    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    @JsonIgnore
    private Set<Comment> comments;
}
